### ISSUE TITLE
Terraform doesn't ask to input variables interactively?

### ISSUE THREAD

**Issue by lawrencepit** [hashicorp/terraform#3481]
Was following the Getting Started at http://www.terraform.io/intro/getting-started/variables.html:

```
UI Input: If you execute terraform plan or apply without doing anything, Terraform will ask you to input the variables interactively.
```

Got this instead:

```
$ terraform plan
There are warnings and/or errors related to your configuration. Please
fix these before continuing.

Errors:

  * 2 error(s) occurred:

* Required variable not set: secret_key
* Required variable not set: access_key
```

The other methods mentioned on that Getting Started page do seem to work.

**Comment by stack72** [comment]
@lawrencepit I believe this is a mistake in the docs. This was added and then reverted. Right now, I believe that input defaults to false so it won't ask for them. 

I usually use a tfvars file to load these types of variables. My tfvars file is then globally ignored by my git config

**Comment by lawrencepit** [comment]
I later noticed the explicit `-input=true` in the [command: apply doc](http://www.terraform.io/docs/commands/apply.html), so I guess a mistake in the docs indeed.

PS. personally I don't like checking in files containing secrets such as aws keys/secrets. I believe Vault is the better solution for that ;-)

**Comment by stack72** [comment]
@phinze Looks like there is a small mistake in the getting started docs:

```
**UI Input:** If you execute `terraform plan` or apply without doing
anything, Terraform will ask you to input the variables interactively.
These variables are not saved, but provides a nice user experience for
getting started with Terraform.
```

This isn't actually the case. By default, input is set to false. I suggest we change to something like

```
**UI Input:** By default, Terraform will *not* ask you to interactively input the variables if you execute `terraform plan` or apply without doing
anything. 
```

**Comment by phinze** [comment]
Hm. This is confusing!

The relevant code is [here](https://github.com/hashicorp/terraform/blob/master/command/meta.go#L196-L201), which says that unset variables should prompt for input __iff__ no variables are specified in other ways (via tfvars or flags).

So, important question: was there a `terraform.tfvars` in the dir when the unexpected lack of prompt happened?

**Comment by lawrencepit** [comment]
I followed the Getting Started page for [Input Variables](https://www.terraform.io/intro/getting-started/variables.html). The first step was `Create another file variables.tf`. I have no `terraform.tfvars` file.

I think what users would expect is that if a var is not defined via the command line, and then not defined in `terraform.tfvars` and finally not defined as a `TF_VAR_*` environment variable, then it falls back to UI input. Although tbh I'd prefer if by default it did not fall back to UI input (because I'd prefer automation without providing a -input=false each time). If you want the UI input fallback you'd have to be explicit by providing `-input=true` via the command line. Which btw also doesn't seem to work for me:

```
$ terraform plan -input=true
There are warnings and/or errors related to your configuration. Please
fix these before continuing.

Errors:

  * 1 error(s) occurred:

* Required variable not set: foo
```

See also http://www.terraform.io/docs/commands/plan.html#_input_true

**Comment by aviflax** [comment]
New Terraform user here, and I’ve also been confused by the discrepancy between the Getting Started guide and the current behavior of the program. So I agree that either the behavior should be changed to match the docs, or the docs should be changed to match the behavior.

----

I’m _also_ seeing the same problem as @lawrencepit wherein the `-input` flag does’t seem to work at all:

```
$ terraform plan -input=true
There are warnings and/or errors related to your configuration. Please
fix these before continuing.

Errors:

  * 3 error(s) occurred:

* Required variable not set: environment
* Required variable not set: access_key
* Required variable not set: secret_key
```

So that should probably be fixed as well.

**Comment by phinze** [comment]
I believe this should be fixed in Terraform 0.6.7 by #3843 and #4017, which cause the input prompt behavior to be consistent across commands. Thanks to @jen20 for the fixes there!
